src/Microsoft.Performance.SDK.Runtime/Options/PluginOptionsRegistry.cs

```diff
@@ -0,0 +1,144 @@
+    ///         <item>
+    ///             Any registered <see cref="PluginOption"/> instances that have a <see cref="Guid"/> which corresponds to a <see cref="PluginOptionDto"/>
+    ///             of the same type in the given <see cref="PluginOptionsDto"/> will either
+    ///             <list type="bullet">
+    ///                 <item>
+    ///                     Have its <see cref="PluginOption{T}.CurrentValue"/> set to the DTO's serialized value if the DTO's <see cref="PluginOptionDto.IsDefault"/>
+    ///                     is <c>false</c>.
+    ///                 </item>
+    ///                 <item>
+    ///                     Be reset to the <see cref="PluginOption{T}"/>'s default state if the DTO's <see cref="PluginOptionDto.IsDefault"/>
+    ///                     is <c>true</c>, disregarding the serialized value.
+    ///                 </item>
+    ///             </list>
+    ///         </item>
+    ///         <item>
+    ///             Any registered <see cref="PluginOption"/> instances that do not have a corresponding <see cref="PluginOptionDto"/> in the given
+    ///             <see cref="PluginOptionsDto"/> will be left unchanged.
+    ///         </item>
+    ///         <item>
+    ///             Any serialized <see cref="PluginOptionDto"/> instances in the given <see cref="PluginOptionsDto"/> that do not have a corresponding
+    ///             registered <see cref="PluginOption"/> instance will be ignored.
+    ///         </item>
+    ///     </list>
+    /// </summary>
+    /// <param name="dto">
+    ///     The <see cref="PluginOptionsDto"/> from which to update the state of the <see cref="PluginOption"/> instances in this registry.
+    /// </param>
+    internal void UpdateFromDto(PluginOptionsDto dto)
+    {
+        ApplyBooleanDtos(dto.BooleanOptions);
+        ApplyFieldDtos(dto.FieldOptions);
+        ApplyFieldArrayDtos(dto.FieldArrayOptions);
+    }
+
+    private void ApplyFieldArrayDtos(IReadOnlyCollection<FieldArrayPluginOptionDto> dtoFieldArrayOptions)
+    {
+        ApplyT<FieldArrayOption, FieldArrayPluginOptionDto>(
+            dtoFieldArrayOptions,
+            (option, dto) => { option.CurrentValue = dto.Value; });
+    }
+
+    private void ApplyFieldDtos(IReadOnlyCollection<FieldPluginOptionDto> dtoFieldOptions)
+    {
+        ApplyT<FieldOption, FieldPluginOptionDto>(
+            dtoFieldOptions,
+            (option, dto) => { option.CurrentValue = dto.Value; });
+    }
+
+    private void ApplyBooleanDtos(IReadOnlyCollection<BooleanPluginOptionDto> dtoBooleanOptions)
+    {
+        ApplyT<BooleanOption, BooleanPluginOptionDto>(
+            dtoBooleanOptions,
+            (option, dto) => { option.CurrentValue = dto.Value; });
+    }
+
+    private void ApplyT<T, TDTO>(
+        IReadOnlyCollection<TDTO> dtoOptions,
+        Action<T, TDTO> applySaved)
+        where T : PluginOption
+        where TDTO : PluginOptionDto
+    {
+        foreach (var dto in dtoOptions)
```

Comment:
Enclose foreach with { } ?

Reply:
Thanks, my IDE auto-formatted this file.